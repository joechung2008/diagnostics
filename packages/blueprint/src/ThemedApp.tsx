import App from "./App";
import { useSystemTheme } from "./useSystemTheme";
import { BlueprintProvider } from "@blueprintjs/core";

const ThemedApp: React.FC = () => {
  const theme = useSystemTheme();

  return (
    <BlueprintProvider>
      <div className={theme === "dark" ? "bp6-dark" : undefined}>
        <App />
      </div>
    </BlueprintProvider>
  );
};

export default ThemedApp;
