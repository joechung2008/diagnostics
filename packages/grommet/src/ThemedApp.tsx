import { Grommet } from "grommet";
import { grommet, dark } from "grommet/themes";
import React, { useEffect, useState } from "react";

interface ThemedAppProps {
  children: React.ReactNode;
}

const ThemedApp: React.FC<ThemedAppProps> = ({ children }) => {
  const [theme, setTheme] = useState(grommet);

  useEffect(() => {
    const mediaQuery = window.matchMedia("(prefers-color-scheme: dark)");

    const handleChange = (e: MediaQueryListEvent) => {
      setTheme(e.matches ? dark : grommet);
    };

    // Set initial theme
    setTheme(mediaQuery.matches ? dark : grommet);

    // Listen for changes
    mediaQuery.addEventListener("change", handleChange);

    return () => {
      mediaQuery.removeEventListener("change", handleChange);
    };
  }, []);

  return <Grommet theme={theme}>{children}</Grommet>;
};

export default ThemedApp;
