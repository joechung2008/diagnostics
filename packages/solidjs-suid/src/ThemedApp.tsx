import { ThemeProvider, createTheme } from "@suid/material";
import { createEffect, createMemo, createSignal, onCleanup } from "solid-js";
import App from "./App";

export default function ThemedApp() {
  const getSystemMode = (): "dark" | "light" =>
    window.matchMedia("(prefers-color-scheme: dark)").matches
      ? "dark"
      : "light";
  const [mode, setMode] = createSignal<"dark" | "light">(getSystemMode());

  createEffect(() => {
    const media = window.matchMedia("(prefers-color-scheme: dark)");
    const handler = (e: MediaQueryListEvent) =>
      setMode(e.matches ? "dark" : "light");
    media.addEventListener("change", handler);
    onCleanup(() => media.removeEventListener("change", handler));
  });

  const theme = createMemo(() =>
    createTheme({
      palette: {
        mode: mode(),
      },
    })
  );

  createEffect(() => {
    document.body.style.backgroundColor = theme().palette.background.default;
  });

  return (
    <ThemeProvider theme={theme}>
      <App />
    </ThemeProvider>
  );
}
