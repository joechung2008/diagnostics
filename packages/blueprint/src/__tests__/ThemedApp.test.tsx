import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import ThemedApp from "../ThemedApp";

describe("ThemedApp", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<ThemedApp />);
    expect(asFragment()).toMatchSnapshot();
  });
});
