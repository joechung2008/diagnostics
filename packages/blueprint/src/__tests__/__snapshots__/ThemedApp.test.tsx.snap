// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ThemedApp > matches snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
